python/ray/train/predictor.py

```diff
@@ -133,34 +133,47 @@ def _set_cast_tensor_columns(self):
         """
         self._cast_tensor_columns = True
 
-    def predict(self, data: DataBatchType, **kwargs) -> DataBatchType:
+    def predict(
+        self, data: DataBatchType, batch_format: Optional[str] = None, **kwargs
+    ) -> DataBatchType:
         """Perform inference on a batch of data.
 
         Args:
             data: A batch of input data of type ``DataBatchType``.
+            batch_format: The format of the input batch.
```

Comment:
can we just identify this based on the type of `data` using the `TYPE_TO_ENUM` defined up top? I think we should just choose the path with least conversion.

Reply:
I see, so the preprocessor is also involved. Can we do something like this instead?

```
preprocessed_data = self._preprocessor.transform_batch(data)
try:
    batch_format = TYPE_TO_ENUM[type(preprocessed_data)]
except KeyError:
    raise RuntimeError("Invalid input type")
if batch_format == BatchFormat.PANDAS:
    predictions = self._predict_pandas(preprocessed_data, **kwargs)
    return convert_pandas_to_batch_type(predictions, TYPE_TO_ENUM[type(data)])
elif batch_format == BatchFormat.NUMPY:
    predictions = self._predict_numpy(preprocessed_data, **kwargs)
    return convert_numpu_to_batch_type(predictions, TYPE_TO_ENUM[type(data)])
```

Reply:
Basically we delegate to the preprocessor to choose the best path for preprocessing. And then whatever the type is of the preprocessed data, we choose the appropriate path for predictions.

Users don't need to think about the batch format for prediction.

Reply:
I see what you meant and actually that's the first thing i tried when implementing this PR too, but `TYPE_TO_ENUM` has an entry with `pa_table: DataType.ARROW` with a bunch of unit tests scattered around our codebase, some of them are quite legit too based on the way there're constructed.

So for arrow data format it's inevitable we need to put the conversion code for it somewhere; Inline it within this function in favor of removing the arg completely worth a try.

Reply:
Hmm not sure I understand- can we just remove the arrow from the TYPE_TO_ENUM dict if it's not supported?